src/praisonai-train/praisonai_train/train/llm/trainer.py

```
@@ -464,12 +679,23 @@ def inference(self, instruction, input_text):
-            model_name=self.config["output_dir"],
-            max_seq_length=2048,
-            dtype=None,
-            load_in_4bit=self.config["load_in_4bit"],
+        # Reload with the SAME precision/quantization the model was trained under,
+        # otherwise an 8-bit (or full-precision) model gets reloaded in 4-bit — a
+        # different memory footprint that can OOM or silently change behaviour.
+        dtype_cfg = self.config.get("dtype")
+        load_kwargs = dict(
+            model_name=self.config.get("final_model_dir", "lora_model"),
+            max_seq_length=self.config.get("max_seq_length", 2048),
+            dtype=getattr(torch, dtype_cfg) if isinstance(dtype_cfg, str) else None,
```

Ruff (0.15.21)

[error] Undefined name `torch`

(F821)